api/datasets.go

```diff
@@ -279,6 +279,10 @@ func (h *DatasetHandlers) getHandler(w http.ResponseWriter, r *http.Request) {
 			NoHistoryErrResponse(w)
 			return
 		}
+		if err == lib.ErrNotLinkedToFilesystem {
+			NoFsiPathErrResponse(w)
```

Comment:
I don't think we need a custom function for this. How about this instead:
```go
if err == fsi.ErrNoLink {
  util.WriteErrResponse(w, http.StatusUnprocessableEntity, err)
  return
}
```

Reply:
Dope.

Should I also refactor the NoHistoryErrResponse func? It's where I took the pattern

Reply:
that'd be lovely!